packages/video_player/video_player_avfoundation/example/ios/RunnerTests/VideoPlayerTests.m

```diff
@@ -176,6 +178,250 @@ - (void)testSeekToInvokesTextureFrameAvailableOnTextureRegistry {
+      makeWithUrl:@"https://storage.googleapis.com/exoplayer-test-media-1/mkv/"
+                  @"android-screens-lavf-56.36.100-aac-avc-main-1280x720.mkv"];
+
+  FLTIsSupportedMessage *response = [videoPlayerPlugin isCacheSupportedForNetworkMedia:message
+                                                                                 error:&error];
+  XCTAssertNil(error);
+  XCTAssertFalse(response.isSupported.boolValue);
+}
+
+- (void)testCantCacheWEBM {
+  NSObject<FlutterTextureRegistry> *mockTextureRegistry =
+      OCMProtocolMock(@protocol(FlutterTextureRegistry));
+  NSObject<FlutterPluginRegistry> *registry =
+      (NSObject<FlutterPluginRegistry> *)[[UIApplication sharedApplication] delegate];
+  NSObject<FlutterPluginRegistrar> *registrar = [registry registrarForPlugin:@"CantCacheWEBM"];
+  NSObject<FlutterPluginRegistrar> *partialRegistrar = OCMPartialMock(registrar);
+  OCMStub([partialRegistrar textures]).andReturn(mockTextureRegistry);
+  FLTVideoPlayerPlugin *videoPlayerPlugin =
+      (FLTVideoPlayerPlugin *)[[FLTVideoPlayerPlugin alloc] initWithRegistrar:partialRegistrar];
+
+  FlutterError *error;
+  [videoPlayerPlugin initialize:&error];
+  XCTAssertNil(error);
+
+  FLTIsCacheSupportedMessage *message = [FLTIsCacheSupportedMessage
+      makeWithUrl:@"https://storage.googleapis.com/exoplayer-test-media-1/gen-3/screens/"
+                  @"dash-vod-single-segment/video-vp9-360.webm"];
+
+  FLTIsSupportedMessage *response = [videoPlayerPlugin isCacheSupportedForNetworkMedia:message
+                                                                                 error:&error];
+  XCTAssertNil(error);
+  XCTAssertFalse(response.isSupported.boolValue);
+}
+
+- (void)testCantCacheMP3 {
+  NSObject<FlutterTextureRegistry> *mockTextureRegistry =
+      OCMProtocolMock(@protocol(FlutterTextureRegistry));
+  NSObject<FlutterPluginRegistry> *registry =
+      (NSObject<FlutterPluginRegistry> *)[[UIApplication sharedApplication] delegate];
+  NSObject<FlutterPluginRegistrar> *registrar = [registry registrarForPlugin:@"CantCacheMP3"];
+  NSObject<FlutterPluginRegistrar> *partialRegistrar = OCMPartialMock(registrar);
+  OCMStub([partialRegistrar textures]).andReturn(mockTextureRegistry);
+  FLTVideoPlayerPlugin *videoPlayerPlugin =
+      (FLTVideoPlayerPlugin *)[[FLTVideoPlayerPlugin alloc] initWithRegistrar:partialRegistrar];
+
+  FlutterError *error;
+  [videoPlayerPlugin initialize:&error];
+  XCTAssertNil(error);
+
+  FLTIsCacheSupportedMessage *message = [FLTIsCacheSupportedMessage
+      makeWithUrl:@"https://storage.googleapis.com/exoplayer-test-media-0/play.mp3"];
+
+  FLTIsSupportedMessage *response = [videoPlayerPlugin isCacheSupportedForNetworkMedia:message
+                                                                                 error:&error];
+  XCTAssertNil(error);
+  XCTAssertFalse(response.isSupported.boolValue);
+}
+
+- (void)testCantCachOGG {
```

Comment:
I think 1-2 common unsupported formats should be good enough - you can't enumerate all of them anyways. 

May make sense to put them into the same test function.